optimum/intel/openvino/modeling_sentence_transformers.py

```diff
@@ -0,0 +1,180 @@
+import warnings
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from types import MethodType
+from typing import Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from huggingface_hub.constants import HUGGINGFACE_HUB_CACHE
+from sentence_transformers import SentenceTransformer
+from transformers import AutoTokenizer, PretrainedConfig
+from transformers.file_utils import add_start_docstrings
+
+from ...exporters.openvino import main_export
+from .configuration import OVConfig, OVWeightQuantizationConfig
+from .modeling import MODEL_START_DOCSTRING, OVModel
+
+
+@add_start_docstrings(
+    """
+    OpenVINO Model for feature extraction tasks for Sentence Transformers.
+    """,
+    MODEL_START_DOCSTRING,
+)
+class OVModelForSentenceTransformer(OVModel):
```

Comment:
For transformers we have `AutoModelForXxx` -> `OVModelForXxx`
For diffusers we have  `XxxPipeline` -> `OVXxxPipeline`
For sentence transformers what do you think about `SentenceTransformer` -> `OVSentenceTransformer` ?


```suggestion
class OVSentenceTransformer(OVModel):
```
also cc @eaidova